applications/qserv-kafka/templates/metrics-cronjob.yaml

```diff
@@ -0,0 +1,95 @@
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: periodic-metrics
+  labels:
+    {{- include "qserv-kafka.labels" . | nindent 4 }}
+spec:
+  schedule: {{ .Values.periodicMetrics.schedule | quote }}
+  concurrencyPolicy: "Forbid"
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels:
+            {{- include "qserv-kafka.selectorLabels" . | nindent 12 }}
+            app.kubernetes.io/component: "periodic-metrics"
+            qserv-kafka-redis-client: "true"
+        spec:
+          restartPolicy: "Never"
+          containers:
+            - name: periodic-metrics
+              command: ["qservkafka_publish_periodic_metrics"]
+              env:
+                - name: "KAFKA_BOOTSTRAP_SERVERS"
+                  valueFrom:
+                    secretKeyRef:
+                      name: "qserv-kafka-access"
+                      key: "bootstrapServers"
+                - name: "KAFKA_CLIENT_CERT_PATH"
+                  value: "/etc/qserv-kafka/user.crt"
+                - name: "KAFKA_CLIENT_KEY_PATH"
+                  value: "/etc/qserv-kafka/user.key"
+                - name: "KAFKA_CLUSTER_CA_PATH"
+                  value: "/etc/qserv-kafka/ca.crt"
+                - name: "KAFKA_SECURITY_PROTOCOL"
+                  valueFrom:
+                    secretKeyRef:
+                      name: "qserv-kafka-access"
+                      key: "securityProtocol"
+                - name: "QSERV_KAFKA_GAFAELFAWR_TOKEN"
+                  valueFrom:
+                    secretKeyRef:
+                      name: "qserv-kafka-token"
+                      key: "token"
+                - name: "QSERV_KAFKA_QSERV_DATABASE_PASSWORD"
+                  valueFrom:
+                    secretKeyRef:
+                      name: "qserv-kafka"
+                      key: "qserv-password"
+                - name: "QSERV_KAFKA_REDIS_PASSWORD"
+                  valueFrom:
+                    secretKeyRef:
+                      name: "qserv-kafka"
+                      key: "redis-password"
+                {{- if .Values.config.qservRestUsername }}
+                - name: "QSERV_KAFKA_QSERV_REST_PASSWORD"
+                  valueFrom:
+                    secretKeyRef:
+                      name: "qserv-kafka"
+                      key: "qserv-password"
+                {{- end }}
+              envFrom:
+                - configMapRef:
+                    name: "qserv-kafka"
```

Comment:
I'm sorry to make you suffer for my laziness, but we're now repeating this stuff three times, so it really should be moved into a helper macro in `_helpers.tpl` similar to `gafaelfawr.envVars` in the Gafaelfawr chart.

Reply:
The periodic worker doesn't need most of this, it only needs redis connection config, metrics config, and a queue name to monitor. I can split the config out here (and in the app code) in a subsequent PR if that's OK with you. For now, I'll pull this config into a template and use it for all three components (frontend, backend, periodic metrics).

Reply:
Oh, in that case, whatever is easier for you! I thought you'd need all of it because you're loading the application config so it has to pass validation.